File: src/JavaScript/easy/array/problem1313.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */


var decompressRLElist = function(nums) {
    let answer = []
    for(let i = 0; i < nums.length; i+=2){
        let arr = [nums[i], nums[i + 1]]
        for(let k = 1; k <= arr[0]; k++){
            answer.push(arr[1])
        }
    }
    return answer
};

// let nums = [1,2,3,4]
let nums = [1,1,2,3]

decompressRLElist(nums)

File: src/Python/easy/array/problem1313.py
from typing import List

class Solution:
    def decompressRLElist(self, nums: List[int]) -> List[int]:
        answer = []
        for i in range(0,len(nums), 2):
            arr = [nums[i], nums[i + 1]]
            for k in range(0, arr[0]):
                answer.append(arr[1])
        return answer



nums = [1,2,3,4]

solution = Solution()
solution.decompressRLElist(nums)